packages/core/src/sync-realtime/index.ts

```diff
@@ -674,10 +675,13 @@ export const createRealtimeSync = (
 
       for (const filter of logFilters) {
         if (isLogFilterMatched({ filter, block, log })) {
-          // TODO: includeTransactionReceipt
-          // return early if includeTransactionReceipt, otherwise continue checking filters
           requiredTransactions.add(log.transactionHash);
           isMatched = true;
+          if (shouldGetTransactionReceipt(filter)) {
+            requiredTransactionReceipts.add(log.transactionHash);
+            // skip to next log
+            break;
```

Comment:
If I'm reading correctly, this `break` skips checking this log against all subsequent filters. I guess this is fine because we've already definitely added this `log.transactionHash` to `requiredTransactions` and `requiredTransactionReceipts`, and that's the only side-effect we care about in this loop?

Control flow is a bit difficult to follow, but looks correct to me.

Reply:
Yes, @khaidarkairbek can you take a look as well?

Reply:
Yes, there is no need to iterate further if we have already included the log and added the tx receipt. These are the only side effects so far.